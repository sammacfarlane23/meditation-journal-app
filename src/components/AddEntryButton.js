import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlus } from '@fortawesome/free-solid-svg-icons';
import EntryModal from './EntryModal';
import { startCreateEntry } from '../actions/entries';

export default () => {
  const [isModalOpen, setIsModalOpen] = useState(false);
  const dispatch = useDispatch();

  const showModal = () => {
    setIsModalOpen(true);
  };

  const onSubmit = (entry) => {
    dispatch(startCreateEntry(entry));
    setIsModalOpen(false);
  };

  const closeModal = () => {
    setIsModalOpen(false);
  };

  return (
    <div className='button-position'>
      {!isModalOpen && (
        <button className='button--big circular-container' onClick={showModal}>
          <FontAwesomeIcon icon={faPlus} size='2x' />
        </button>
      )}
      <EntryModal
        showModal={isModalOpen}
        onSubmit={onSubmit}
        closeModal={closeModal}
      />
    </div>
  );
};
