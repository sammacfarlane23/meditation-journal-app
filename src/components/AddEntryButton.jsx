import React from 'react';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlus } from '@fortawesome/free-solid-svg-icons';
import EntryModal from './EntryModal';
import { startCreateEntry } from '../actions/entries';

export class AddEntryButton extends React.Component {
  state = { show: false };

  showModal = () => {
    this.setState(() => ({ show: true }));
  };

  onSubmit = (entry) => {
    this.props.startCreateEntry(entry);
    this.setState({ show: false });
  };

  closeModal = () => {
    this.setState({ show: false });
  };
  render() {
    return (
      <div className='button-position'>
        {!this.state.show && (
          <button
            className='button--big circular-container'
            onClick={this.showModal}
          >
            <FontAwesomeIcon icon={faPlus} size='2x' />
          </button>
        )}
        <EntryModal
          showModal={this.state.show}
          onSubmit={this.onSubmit}
          closeModal={this.closeModal}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startCreateEntry: (entry) => dispatch(startCreateEntry(entry)),
});

export default connect(undefined, mapDispatchToProps)(AddEntryButton);
