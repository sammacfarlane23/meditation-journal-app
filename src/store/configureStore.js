import { createStore, combineReducers, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import authReducer from "../reducers/auth";
import entriesReducer from "../reducers/entries";
import filtersReducer from "../reducers/filters";

const configuredStore = () => {
  const store = createStore(
    combineReducers({
      entries: entriesReducer,
      auth: authReducer,
      filters: filtersReducer,
    }),
    applyMiddleware(thunk)
  );

  return store;
};

export default configuredStore;
