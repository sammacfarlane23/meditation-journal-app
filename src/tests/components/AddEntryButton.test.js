import React from 'react';
import { screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { renderWithStore } from '../store/mockStore';
import AddEntryButton from '../../components/AddEntryButton';

test('should render AddEntryButton correctly with isModalOpen false', () => {
  const component = renderWithStore(<AddEntryButton />);
  expect(component).toMatchSnapshot();
});

test('Clicking the button should set isModalOpen to true', () => {});
