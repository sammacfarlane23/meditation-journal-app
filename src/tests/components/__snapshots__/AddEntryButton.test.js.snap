// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddEntryButton correctly with isModalOpen false 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
        <div
            class="button-position"
        >
            <button
                class="button--big circular-container"
            >
                <div>
                    <svg
                        aria-hidden="true"
                        focusable="false"
                        data-prefix="fas"
                        data-icon="plus"
                        class="svg-inline--fa fa-plus fa-w-14 fa-2x "
                        role="img"
                        xmlns="http://www.w3.org/2000/svg"
                        viewBox="0 0 448 512"
                    >
                        <path
                            fill="currentColor"
                            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                        />
                    </svg>
                </div>
            </button>
        </div>
    </div>
    <div
        class="ReactModalPortal"
    />
</body>,
  "container": <div>
    <div
        class="button-position"
    >
        <button
            class="button--big circular-container"
        >
            <div>
                <svg
                    aria-hidden="true"
                    focusable="false"
                    data-prefix="fas"
                    data-icon="plus"
                    class="svg-inline--fa fa-plus fa-w-14 fa-2x "
                    role="img"
                    xmlns="http://www.w3.org/2000/svg"
                    viewBox="0 0 448 512"
                >
                    <path
                        fill="currentColor"
                        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                    />
                </svg>
            </div>
        </button>
    </div>
</div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
